### PR TITLE
Use snprintf instead of sprintf.

### PR DESCRIPTION
**Pull request by toregge**
@baldersheim : please review

### CODE REVIEW

File: vespalib/src/vespa/fastlib/io/bufferedfile.cpp [hunk 1]
```diff
@@ -186,7 +186,7 @@ Fast_BufferedFile::SetPosition(const int64_t s)
             diff = _filepos - s;
             if ( !(((diff > 0l) || ((diff == 0l) && (_fileleft == 0l))) && (diff <= static_cast<int64_t>(_buf.size())))) {
                 char tmp[8196];
```
Comment (baldersheim) [inline vespalib/src/vespa/fastlib/io/bufferedfile.cpp:188]: 64 should be fine

File: vespalog/src/logger/runserver.cpp [hunk 2]
```diff
@@ -319,7 +319,7 @@ int loop(const char *svc, char * const * run)
             LOG(debug, "got signal %d, sending to pid %d",
                 (int)lastsig, (int)child);
             char why[256];
```
Comment (baldersheim) [inline vespalog/src/logger/runserver.cpp:321]: 32 should be fine

**Review by baldersheim — APPROVED** [review]

**Review by baldersheim — APPROVED** [review]

File: vespalib/src/vespa/fastlib/io/bufferedfile.cpp (hunk 1 above)
Comment (toregge) [reply vespalib/src/vespa/fastlib/io/bufferedfile.cpp]: Adjusted.

File: vespalog/src/logger/runserver.cpp (hunk 2 above)
Comment (toregge) [reply vespalog/src/logger/runserver.cpp]: Adjusted.

**Review by baldersheim — APPROVED** [review]

**Pull request merged**